src/extension.ts

```diff
@@ -150,10 +145,35 @@ export function activate(context: ExtensionContext) {
 
 	const enable: boolean = workspace.getConfiguration('inmanta').ls.enabled
 
+	function createVenvIfNotExists() {
+		const pp: string = workspace.getConfiguration('inmanta').pythonPath;
+		if (pp && fs.existsSync(pp)) {
+			window.showInformationMessage(`Using Python Interpreter at ${pp}`);
+			return pp;
+		} else {
+			window.showInformationMessage(`No Python3 interpreter found at "${pp}". Falling back to default virtual environment`);
+		}
+		const venvDir = Uri.joinPath(context.globalStorageUri, ".env").fsPath;
+		const venvPath = Uri.joinPath(context.globalStorageUri, ".env", "bin", "python3").fsPath;
+
+		if (!fs.existsSync(venvDir)) {
+			const venvProcess = cp.spawnSync("python3", ["-m", "venv", venvDir ]);
+			if (venvProcess.status != 0) {
+				window.showErrorMessage(`Virtual env creation at ${venvDir} failed with code ${venvProcess.status}, ${venvProcess.stderr}`);
+			}
+			const child = cp.spawnSync(venvPath, ["-m", "pip", "install", "--pre", "inmantals"]);
```

Comment:
Why `--pre`?

Reply:
I see now it was there before your changes as well.

Reply:
Yes, but now that you mention it, I'm not sure if it's necessary

Reply:
good point, should probably be removed, those who want --pre can do it by hand

Reply:
(perhaps both pip-install's have to go into one method)